interface/super/edit_globals.php

```diff
@@ -187,55 +188,73 @@ function checkBackgroundServices(){
   $auditLogStatusFieldOld = $GLOBALS['enable_auditlog'];
 
+  /*
+   * Compare form values with old database values.
+   * Only save if values differ. Improves speed.
+   */
+  
+  // Get all the globals from DB
+  $old_globals = $GLOBALS['adodb']['db']->getAssoc('SELECT gl_name, gl_index, gl_value FROM `globals` ORDER BY gl_name, gl_index',false,true);
+  
   $i = 0;
   foreach ($GLOBALS_METADATA as $grpname => $grparr) {
     foreach ($grparr as $fldid => $fldarr) {
       list($fldname, $fldtype, $flddef, $flddesc) = $fldarr;
 	  if($fldtype == 'pwd'){
-	  $pass = sqlQuery("SELECT gl_value FROM globals WHERE gl_name = '$fldid'");
-	  $fldvalueold = $pass['gl_value'];
+        $pass = sqlQuery("SELECT gl_value FROM globals WHERE gl_name = '$fldid'");
+        $fldvalueold = $pass['gl_value'];
 	  }
-      sqlStatement("DELETE FROM globals WHERE gl_name = '$fldid'");
 
+      /* Multiple choice fields - do not compare , overwrite */
       if (!is_array($fldtype) && substr($fldtype, 0, 2) == 'm_') {
         if (isset($_POST["form_$i"])) {
           $fldindex = 0;
+          
+          sqlStatement("DELETE FROM globals WHERE gl_name = ?", array( $fldid ) );
+          
           foreach ($_POST["form_$i"] as $fldvalue) {
             $fldvalue = formDataCore($fldvalue, true);
-            sqlStatement("INSERT INTO globals ( gl_name, gl_index, gl_value ) " .
-              "VALUES ( '$fldid', '$fldindex', '$fldvalue' )");
+            sqlStatement('INSERT INTO `globals` ( gl_name, gl_index, gl_value ) VALUES ( ?,?,?)', array( $fldid, $fldindex, $fldvalue )  );
             ++$fldindex;
           }
         }
       }
       else {
+        /* check value of single field. Don't update if the database holds the same value */
         if (isset($_POST["form_$i"])) {
           $fldvalue = formData("form_$i", "P", true);
         }
         else {
           $fldvalue = "";
         }
-        if($fldtype=='pwd') $fldvalue = $fldvalue ? SHA1($fldvalue) : $fldvalueold;
-        if(fldvalue){
-          // Need to force enable_auditlog_encryption off if the php mycrypt module
-          // is not installed.
-          if ( $force_off_enable_auditlog_encryption && ($fldid  == "enable_auditlog_encryption") ) {
-            error_log("OPENEMR ERROR: UNABLE to support auditlog encryption since the php mycrypt module is not installed",0);
-            $fldvalue=0;
-          }
-          // special treatment for some vars
-          switch ($fldid) {
-            case 'first_day_week':
-              // update PostCalendar config as well
-              sqlStatement("UPDATE openemr_module_vars SET pn_value = ? WHERE pn_name = 'pcFirstDayOfWeek'", array($fldvalue));
-              break;
+        if($fldtype=='pwd') $fldvalue = $fldvalue ? SHA1($fldvalue) : $fldvalueold; // TODO: salted passwords?
+
+        if( !isset( $old_globals[$fldid]) ) {
+          // We rely on the fact that set of keys in globals.inc === set of keys in `globals`  table!
```

Comment:
In any case, shouldn't allow creating global settings that are not included in globals.inc.